harness/determined/deploy/gcp/gcp.py

```diff
@@ -348,3 +353,57 @@ def set_gcp_credentials_env(tf_vars: Dict) -> None:
 def wait_for_master(configs: Dict, env: Dict, timeout: int = 300) -> None:
     master_url = terraform_output(configs, env, "Web-UI")
     healthcheck.wait_for_master_url(master_url, timeout)
+
+
+def check_or_create_gcsbucket(project_id: str, keypath: Optional[str] = None) -> None:
+    if keypath:
+        os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = keypath
+    bucket_name = project_id + "-determined-deploy"
+    try:
+        storage_service = googleapiclient.discovery.build("storage", "v1")
+    except DefaultCredentialsError:
+        print(
+            colored("Unable to locate GCP credentials.", "red"),
+            "Please set %s or explicitly create credentials"
+            % colored("GOOGLE_APPLICATION_CREDENTIALS", "yellow"),
+            "and re-run the application. ",
+            "For more information, please see",
+            "https://docs.determined.ai/latest/sysadmin-deploy-on-gcp/install-gcp.html#credentials",
+            "and",
+            "https://cloud.google.com/docs/authentication/getting-started",
+        )
+        sys.exit(1)
+
+    try:
+        storage_service.buckets().get(bucket=bucket_name).execute()
+    except HttpError as err:
+        if err.resp.status == 404:
+            request_body = {
+                "name": bucket_name,
+            }
+            storage_service.buckets().insert(project=project_id, body=request_body).execute()
+        else:
+            raise
+
+
+def list_clusters(
+    bucket_name: str, json_bool: Optional[bool] = False, yaml_bool: Optional[bool] = False
```

Comment:
we don't usually do hungarian notation (except for something like file or directory paths), so perhaps:
```suggestion
    bucket_name: str, json: bool = False, yaml: bool = False
```

OR, since they are mutually exclusive anyway, have a string which is one of `"table"|"json"|"yaml"`:
```suggestion
    bucket_name: str, format: str = "table"
```

Reply:
Done!